source/MaterialXCore/Node.cpp

```diff
@@ -647,22 +647,38 @@ void NodeGraph::setNodeDef(ConstNodeDefPtr nodeDef)
     }
 }
 
-InputPtr Node::addInputFromNodeDef(const string& name)
+InputPtr Node::addInputFromNodeDef(const string& name, NodeDefPtr fromNodeDef)
 {
+    const StringSet valueAttributes = { ValueElement::TYPE_ATTRIBUTE,
+                                        ValueElement::VALUE_ATTRIBUTE };
+   
     InputPtr nodeInput = getInput(name);
     if (!nodeInput)
     {
-        NodeDefPtr nodeDef = getNodeDef();
+        NodeDefPtr nodeDef = fromNodeDef ? fromNodeDef : getNodeDef();
         InputPtr nodeDefInput = nodeDef ? nodeDef->getActiveInput(name) : nullptr;
         if (nodeDefInput)
         {
             nodeInput = addInput(nodeDefInput->getName());
-            nodeInput->copyContentFrom(nodeDefInput);
+            nodeInput->copyContentFrom(nodeDefInput, valueAttributes);
```

Comment:
In this situation, perhaps we don't even need these changes to the API of Element::copyContentFrom, and we could instead copy these two attributes in the most direct and efficient way:

```
const string& nodeDefType = nodeDefInput->getType();
const string& nodeDefValueString = nodeDefInput->getValueString();
nodeInput->setType(nodeDefType);
nodeInput->setValueString(nodeDefValueString);
```

This would avoid the additional overhead of checking for children of the input element, and would avoid adding new checks for existing users of Element::copyContentFrom.